components/automate-ui/src/app/ui.component.ts

```diff
@@ -61,7 +62,14 @@ export class UIComponent implements OnInit {
   ngOnInit(): void {
     this.router.events.pipe(
         filter(event => event instanceof ActivationStart)
-    ).subscribe((event: any) => this.renderNavbar = !event.snapshot.data.hideNavBar);
+    ).subscribe((event: ActivationStart) => this.renderNavbar = !event.snapshot.data.hideNavBar);
+
+    this.router.events.pipe(
+        filter(event => event instanceof NavigationEnd)
+    ).subscribe(() => this.store.dispatch(new GetAllUserPerms()));
```

Comment:
(2) Every time you navigate to a new Angular route (generally equivalent to when you go to a new view) it will re-fetch the introspection data _but only if it has gone stale_ (been around more than 30 seconds). If it is still fresh, no data is fetched.

There are a variety of lifecycle hooks in routing; this particular one (`NavigationEnd`) typically occurs about once per route (minimum 1, maximum 2, from what I have observed). If it is triggered more than once, that is OK--see point (3).